temp_name_lib/src/star_physics/intensity.rs

```diff
@@ -0,0 +1,71 @@
+use nalgebra as na;
+
+use crate::utils::MathErrors;
+
+enum GridPointStatus{
+    NotAvailable,
+    NotLoaded,
+    Loaded,
+}
+
+enum IntensityType{
+    Total,
+    ContinuumOnly
+}
+
+pub struct Intensity {
+    temp_eff:Vec<f64>,
+    log_g:Vec<f64>,
+    lambda:Vec<f64>,
+    file_name:String,//[Ricardo]: I suppose you have different intensity files.
+    
+    temp_eff_index:usize,
+    log_g_index:usize,
+    lambda_index:usize,
+
+    lambda_lower:f64,
+    lambda_upper:f64,
+    limb_darkening_coef:na::Vector4<f64>,
+    grid_status:GridPointStatus,
+    kind:IntensityType,
+}
+
+
+impl Intensity{
+
+    pub fn read_file(){}
+    pub fn read_info(){}
+    
+    pub fn set_wavelength_range(&mut self,lower:f64,upper:f64){
```

Comment:
Why this function? The intensities are read from a file. So the wavelength range is clear from what's in the file.